addons/fleet/fleet.py

```diff
@@ -1,818 +1,725 @@
+    transmission = fields.Selection(selection=[('manual', 'Manual'), ('automatic', 'Automatic')],
+                                    help='Transmission Used by the vehicle')
+    fuel_type = fields.Selection(selection=[('gasoline', 'Gasoline'), ('diesel', 'Diesel'),
+                                            ('electric', 'Electric'), ('hybrid', 'Hybrid')],
+                                 help='Fuel Used by the vehicle')
+    horsepower = fields.Integer()
+    horsepower_tax = fields.Float(string='Horsepower Taxation')
+    power = fields.Integer(help='Power in kW of the vehicle')
+    co2 = fields.Float(string='CO2 Emissions', help='CO2 emissions of the vehicle')
+    image = fields.Binary(related='model_id.make_id.image', string="Logo", store=True)
+    image_medium = fields.Binary(related='model_id.make_id.image_medium', string="Logo (medium)", store=True)
+    image_small = fields.Binary(related='model_id.make_id.image_small', string="Logo (small)", store=True)
+    contract_renewal_due_soon = fields.Boolean(compute='_compute_get_contract_reminder',
+                                               search='_search_contract_renewal_due_soon',
+                                               string='Has Contracts to renew')
+    contract_renewal_overdue = fields.Boolean(compute='_compute_get_contract_reminder',
+                                              search='_search_get_overdue_contract_reminder',
+                                              string='Has Contracts Overdue')
+    contract_renewal_name = fields.Text(compute='_compute_get_contract_reminder', string='Name of contract to renew soon')
+    contract_renewal_total = fields.Integer(compute='_compute_get_contract_reminder',
+                                            string='Total of contracts due or overdue minus one')
+    car_value = fields.Float('Car Value', help='Value of the bought vehicle')
+
+    _sql_constraints = [('unique_chassis_number', 'unique(vin_sn)', 'Same Chassis Number is already exists.')]
+
+    @api.onchange('model_id')
+    def on_change_model(self):
+        self.image = self.model_id.image
+
+    @api.model
+    def create(self, values):
+        vehicle = super(FleetVehicle, self).create(values)
+        vehicle.message_post(body=_('%s %s has been added to the fleet!') % (vehicle.model_id.name,
+                             vehicle.license_plate))
+        return vehicle
+
+
+class FleetVehicleOdometer(models.Model):
+
+    _name = 'fleet.vehicle.odometer'
+    _description = 'Odometer log for a vehicle'
+    _order = 'date desc'
+
+    @api.depends('vehicle_id')
+    def _compute_vehicle_log_name(self):
+        name = self.vehicle_id.name if self.vehicle_id.name else ''
+        if self.date:
+            name = name + ' / ' + str(self.date)
+        self.name = name
+
+    @api.onchange('vehicle_id')
+    def on_change_vehicle(self):
+        self.unit = self.vehicle_id.odometer_unit
```

Comment:
Please check whether this is necessary. The field `unit` is a related field, and onchanges are automatic on those fields.